lib/workers/branch/index.spec.ts

```diff
@@ -946,5 +953,109 @@ describe('workers/branch', () => {
         )
       ).not.toBeUndefined();
     });
+
+    it('executes post-upgrade tasks once when set to branch mode', async () => {
+      const updatedPackageFile: File = {
+        name: 'pom.xml',
+        contents: 'pom.xml file contents',
+      };
+      getUpdated.getUpdatedPackageFiles.mockResolvedValueOnce({
+        updatedPackageFiles: [updatedPackageFile],
+        artifactErrors: [],
+      } as never);
+      npmPostExtract.getAdditionalFiles.mockResolvedValueOnce({
+        artifactErrors: [],
+        updatedArtifacts: [
+          {
+            name: 'yarn.lock',
+            contents: Buffer.from([1, 2, 3]) /* Binary content */,
+          },
+        ],
+      } as never);
+      git.branchExists.mockReturnValueOnce(true);
+      platform.getBranchPr.mockResolvedValueOnce({
+        title: 'rebase!',
+        state: PrState.Open,
+        body: `- [x] <!-- rebase-check -->`,
+      } as never);
+      git.isBranchModified.mockResolvedValueOnce(true);
+      git.getRepoStatus.mockResolvedValueOnce({
+        modified: ['modified_file', 'modified_then_deleted_file'],
+        not_added: [],
+        deleted: ['deleted_file', 'deleted_then_created_file'],
+      } as StatusResult);
+
+      fs.outputFile.mockReturnValue();
+      fs.readFile
+        .mockResolvedValueOnce(Buffer.from('modified file content'))
+        .mockResolvedValueOnce(Buffer.from('this file will not exists'));
+
+      schedule.isScheduledNow.mockReturnValueOnce(false);
+      commit.commitFilesToBranch.mockResolvedValueOnce(null);
+
+      const adminConfig = {
+        allowedPostUpgradeCommands: ['^echo hardcoded-string$'],
+        allowPostUpgradeCommandTemplating: true,
+        trustLevel: 'high',
+      };
+      setAdminConfig(adminConfig);
+
+      const inconfig: BranchConfig = {
+        ...config,
+        postUpgradeTasks: {
+          executionMode: 'branch',
+          commands: ['echo hardcoded-string', 'disallowed task'],
+          fileFilters: [
+            'modified_file',
+            'deleted_file',
+            'deleted_then_created_file',
+            'modified_then_deleted_file',
+          ],
+        },
+        localDir: '/localDir',
+        upgrades: [
+          {
+            ...defaultConfig,
+            depName: 'some-dep-name-1',
+            postUpgradeTasks: {
+              executionMode: 'branch',
+              commands: ['echo hardcoded-string', 'disallowed task'],
+              fileFilters: [
+                'modified_file',
+                'deleted_file',
+                'deleted_then_created_file',
+                'modified_then_deleted_file',
+              ],
+            },
+          } as never,
```

Comment:
please use `partial<BranchUpgradeConfig>(...)` from `/test/utils`. So we have more type safety. So we don't need a never cast here.

Reply:
Yeah I copy-pasted this from the other cases, have updated the once I touched

Reply:
Done, also fixed. There was 1 as never I couldn't remove because the test started failing, someone should probably investigate that. Since it's a test completely unrelated to my changes I've not tried myself.